src/app.js

```diff
@@ -1,41 +1,101 @@
+  const token = JSON.parse(connect.body).value
+  const cookie = `vmware-api-session-id=${token}`
+  let dcObj = {
+    uri : `https://${host}/rest/vcenter/datacenter`,
+    method: 'GET',
+    rejectUnauthorized: false,
+    headers : {
+      'Authorization': `Bearer ${token}`,
+      'Cookie' : `${cookie}`
+    }
+  }
+  return makeReuqest(dcObj)
 }
 
-function powerOn(vmName){
-  return vm.powerOn(vmName);
+async function createDC (action, settings) {
+  let host = action.params.host || settings.host
+  let connect = await initialConnect(action, settings);
+  const token = JSON.parse(connect.body).value
+  const cookie = `vmware-api-session-id=${token}`
+  const dcFolder = action.params.folder || "group-d1"
+  const dcName = action.params.name
+  let dcObj = {
+    url : `https://${host}/rest/vcenter/datacenter`,
+    method: 'POST',
+    rejectUnauthorized : false,
+    headers : {
+      'Cookie' : `${cookie}`,
+      'Content-Type': 'application/json'
+    },
+    
+    body: JSON.stringify({
+      "spec":{
+        "name":`${dcName}`,
+        "folder":`${dcFolder}`
+      }
+    })
+  };
+  return makeReuqest(dcObj)
 }
 
-function powerOff(vmName){
-  return vm.powerOff(vmName);
+
+
+//////////// HELPERS ////////////
+
+async function initialConnect (action, settings) {
+  let host = action.params.host || settings.host
+  let user = action.params.user || settings.user
+  let password = action.params.password || settings.password
+  let httpReq = {
+    uri: `https://${host}/rest/com/vmware/cis/session`,
+    method: 'POST',
+    rejectUnauthorized: false,
+    auth : {
+      user : `${user}`,  //'administrator@vsphere.local',
+      password: `${password}`
+    }
+  }; 
+  return new Promise((resolve,reject)=>{
+    request(httpReq, function (err, response, body) {
+      if(err){
+          return reject(err);
+      }
+      if(response.statusCode < 200 || response.statusCode > 300){cd 
+        return reject(response.message);
+      }
+      var cookieValue = response.headers['set-cookie'];
+      httpReq.headers = {'Cookie': cookieValue};
+      // Remove username-password authentication.
+      httpReq.auth = {};
+      resolve(response)
+    });
+  });
 }
 
-function execPowerOn(action, settings) {
-    return auth.login(action,settings).then(()=>{
-      return findVMByName(action.params.vmName)
-    }).then(vm=>{
-      return powerOn(action.params.vmName);
-    })
- }
-
- function execPowerOff(action, settings) {
-  return auth.login(action,settings).then(()=>{
-    return findVMByName(action.params.vmName)
-  }).then(vm=>{
-    return powerOff(action.params.vmName);
-  })
+function makeReuqest(my_http_options){
+  var x = JSON.stringify(my_http_options)
+  return new Promise((resolve,reject)=>{
+    request(my_http_options, function (err, response, body) {
+      if(err){
+          return reject(err);
+      }
+      if(response.statusCode < 200 || response.statusCode > 300){
+        var b = JSON.stringify(response.body)
```

Comment:
DO NOT use `var` use either `let` or `const` so it is clear if the value is going to change or not